x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceSparseEmbeddingsModel.java

```diff
@@ -102,19 +94,29 @@ public URI uri() {
         return uri;
     }
 
-    private URI createUri() throws URISyntaxException {
+    private URI createUri() throws ElasticsearchStatusException {
         String modelId = getServiceSettings().modelId();
         String modelIdUriPath;
 
         switch (modelId) {
             case ElserModels.ELSER_V2_MODEL -> modelIdUriPath = "ELSERv2";
-            default -> throw new IllegalArgumentException(
-                String.format(Locale.ROOT, "Unsupported model for %s [%s]", ELASTIC_INFERENCE_SERVICE_IDENTIFIER, modelId)
+            default -> throw new ElasticsearchStatusException(
+                String.format(Locale.ROOT, "Unsupported model for %s [%s]", ELASTIC_INFERENCE_SERVICE_IDENTIFIER, modelId),
+                RestStatus.BAD_REQUEST
             );
         }
 
-        return new URI(
-            elasticInferenceServiceComponents().elasticInferenceServiceUrl() + "/api/v1/sparse-text-embeddings/" + modelIdUriPath
-        );
+        try {
+            // TODO, consider transforming the base URL into a URI for better error handling.
+            return new URI(
+                elasticInferenceServiceComponents().elasticInferenceServiceUrl() + "/api/v1/sparse-text-embeddings/" + modelIdUriPath
+            );
+        } catch (URISyntaxException e) {
+            throw new ElasticsearchStatusException(
+                "Failed to create URI for sparse embeddings service: " + e.getMessage(),
```

Comment:
Should we also be a bit more specific here and use the service name/task type constants?